tools/wt_to_mdb_bson.py

```diff
@@ -53,24 +143,67 @@ def decode_data_section(mdb_file_contents, data_index):
         obj = bson.decode_all(byt)[0]
 
         print('Key:\t%s' % key)
-        print('Value:\n\t%s' % (pprint.pformat(obj, indent=1).replace('\n', '\n\t'),))
+        print('Value:\n\t%s' % (print_bson(obj),))
 
-def dump_mdb_file(wtpath, filename):
+# Convert the output of ./wt -r dump -x to bson.
+def wt_dump_to_bson(wt_output):
     # Dump the MongoDB file into hex format.
-    mdb_hex = subprocess.check_output([wtpath, "dump", "-x", "file:" + filename], universal_newlines=True)
-
-    mdb_file_contents = mdb_hex.splitlines()
+    mdb_file_contents = wt_output
     data_index = find_data_section(mdb_file_contents)
     if data_index > 0:
         decode_data_section(mdb_file_contents, data_index)
     else:
         print("Error: No data section was found in the file.")
-        exit()        
+        exit()
+
+# Call the wt util if required.
+def execute_wt(mode, wtpath, filename):
+    if mode == Mode.DUMP:
+        return subprocess.check_output(
+            [wtpath, "-r", "dump", "-x", "file:" + filename], universal_newlines=True).splitlines()
+    elif mode == Mode.VERIFY:
+        return subprocess.check_output(
+            [wtpath, "-r", "verify", "-d", "dump_pages", "file:" + filename], universal_newlines=True).splitlines()
+    else:
+        return subprocess.check_output(
+            [wtpath, "-r", "printlog", "-u", "-x"], universal_newlines=True).splitlines()
```

Comment:
This is a neat shorthand for having the script invoke `wt`. Couple remarks:
* I don't see this using any of the -C to set additional configuration. Specifically saying the WT log is kept in the `journal` directory. And that the log is compressed with snappy.
* I would expect some of these calls (particularly `printlog`) to have GBs of output. And I believe this is buffering all the output into a string before splitting. 20 years ago, python would choke on that growing a string to that size. Maybe things have changed.

Reply:
I added the -C configuration for printlog. I am hoping the many GBs issue isn't a big problem. If people report that issue we can look into a buffered solution.